2020/day3.clj

```diff
@@ -38,52 +54,61 @@
 (def slopes [[1 1] [3 1] [5 1] [7 1] [1 2]])
 
 (comment
-  (do (def sample-input (file->seq "resources/2020/day3/input-sample.txt"))
-      sample-input)
+  (do (def sample-grid (file->seq "resources/2020/day3/input-sample.txt"))
+      sample-grid)
   #_=> ["..##......."
         ...
         ".#..#...#.#"]
 
-  (take-while neg? [-2 0 1])
-  #_=> (-2)
+  (next-location [1 1] [3 1])
+  #_=> [4 2]
 
-  (take 3 (map #(next-location [0 0] [(* 3 %) (* 1 %)])
-               (range)))
-  #_=> ([0 0] [3 1] [6 2])
+  (next-location-step 3 1 2)
+  #_=> [6 2]
 
-  (locations-s 3 [3 1])
+  (->> (locations-s [3 1])
+       (take 3))
   #_=> ([0 0] [3 1] [6 2])
 
-  (mod 10 5)
-  #_=> 0
+  (with-height? [3 1] 10)
+  #_=> true
 
-  (get-in sample-input [0 2])
-  #_=> \#
+  (with-height? [3 1] 1)
+  #_=> false
 
-  (get-in sample-input [0 0])
+  (take-while neg? [-2 0 1])
+  #_=> (-2)
+
+  (nth ["a" "b" "c"] 1)
+  #_=> "b"
+
+  (char-at-location sample-grid [0 0])
   #_=> \.
 
+  (char-at-location sample-grid [2 0])
+  #_=> \#
```

Comment:
These multi-usage examples are good, but they don't need to be separate expressions.

i.e., just `mapv` over these locations

Reply:
Bump